File: http/src/main/java/com/xinzy/http/CookieJarImp.java
package com.xinzy.http;

import android.support.annotation.NonNull;

import java.io.File;
import java.util.List;

import okhttp3.Cookie;
import okhttp3.CookieJar;
import okhttp3.HttpUrl;

/**
 * Created by Xinzy on 2017/6/9.
 */

class CookieJarImp implements CookieJar {
    private CookieStore mCookieStore;

    public CookieJarImp() {
        mCookieStore = new CookieStore.MemoryCookieStore();
    }

    CookieJarImp(@NonNull String dir) {
        File cookieDir = new File(dir, "cookie");
        mCookieStore = new CookieStore.PersistentCookieStore(cookieDir);
    }

    @Override
    public void saveFromResponse(HttpUrl url, List<Cookie> cookies) {
        mCookieStore.save(url, cookies);
    }

    @Override
    public List<Cookie> loadForRequest(HttpUrl url) {
        return mCookieStore.load(url);
    }
}

File: httpkotlin/src/main/java/com/xinzy/http/kotlin/CookieJarImp.kt
package com.xinzy.http.kotlin

import java.io.File

import okhttp3.Cookie
import okhttp3.CookieJar
import okhttp3.HttpUrl

/**
 * Created by Xinzy on 2017/6/9.
 */

internal class CookieJarImp : CookieJar {
    private val mCookieStore: CookieStore

    constructor() {
        mCookieStore = MemoryCookieStore()
    }

    constructor(dir: File) {
        val cookieDir = File(dir, "cookie")
        mCookieStore = PersistentCookieStore(cookieDir)
    }

    override fun saveFromResponse(url: HttpUrl, cookies: List<Cookie>) {
        mCookieStore.save(url, cookies)
    }

    override fun loadForRequest(url: HttpUrl): List<Cookie> {
        return mCookieStore.load(url)
    }
}
